extensions/api/data-management/asset/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/asset/service/AssetServiceImpl.java

```diff
@@ -67,7 +69,10 @@ public ServiceResult<Asset> create(Asset asset, DataAddress dataAddress) {
     public ServiceResult<Asset> delete(String assetId) {
         return transactionContext.execute(() -> {
 
-            var negotiationsOnAsset = contractNegotiationStore.getNegotiationsWithAgreementOnAsset(assetId);
+            var query = QuerySpec.Builder.newInstance()
+                    .filter(List.of(new Criterion("contractAgreement.assetId", "=", assetId)))
```

Comment:
"contractAgreement.assetId" should be a constant. We should also add a test to ensure ContractAgreement#assetId is not changed.

Reply:
What situation should be verified? I mean, where would the assetId change?